solidity/src/hyperkzg/HyperKZGVerifier.pre.sol

```diff
@@ -0,0 +1,249 @@
+            }
+            // IMPORT-YUL ../base/ECPrecompiles.pre.sol
+            function ec_mul(args_ptr) {
+                pop(staticcall(0, 0, 0, 0, 0, 0))
+                revert(0, 0)
+            }
+            // IMPORT-YUL ../base/ECPrecompiles.pre.sol
+            function ec_pairing_x2(args_ptr) -> success {
+                pop(staticcall(0, 0, 0, 0, 0, 0))
+                revert(0, 0)
+            }
+            // IMPORT-YUL ../base/ECPrecompiles.pre.sol
+            function ec_mul_assign(args_ptr, scalar) {
+                pop(staticcall(0, 0, 0, 0, 0, 0))
+                revert(0, 0)
+            }
+            // IMPORT-YUL ../base/ECPrecompiles.pre.sol
+            function calldata_ec_mul_add_assign(args_ptr, c_ptr, scalar) {
+                pop(staticcall(0, 0, 0, 0, 0, 0))
+                revert(0, 0)
+            }
+            // IMPORT-YUL ../base/ECPrecompiles.pre.sol
+            function ec_add_assign(args_ptr, c_ptr) {
+                pop(staticcall(0, 0, 0, 0, 0, 0))
+                revert(0, 0)
+            }
+            // IMPORT-YUL ../base/ECPrecompiles.pre.sol
+            function constant_ec_mul_add_assign(args_ptr, x, y, scalar) {
+                pop(staticcall(0, 0, 0, 0, 0, 0))
+                revert(0, 0)
+            }
+            // IMPORT-YUL ../base/ECPrecompiles.pre.sol
+            function calldata_ec_add_assign(args_ptr, c_ptr) {
+                pop(staticcall(0, 0, 0, 0, 0, 0))
+                revert(0, 0)
+            }
+            // IMPORT-YUL HyperKZGHelpers.pre.sol
+            function run_transcript(com_ptr, v_ptr, w_ptr, transcript_ptr, ell) -> r, q, d {
+                revert(0, 0)
+            }
+            // IMPORT-YUL HyperKZGHelpers.pre.sol
+            function bivariate_evaluation(v_ptr, q, d, ell) -> b {
+                revert(0, 0)
+            }
+            // IMPORT-YUL HyperKZGHelpers.pre.sol
+            function check_v_consistency(v_ptr, r, x, y) {
+                revert(0, 0)
+            }
+            // IMPORT-YUL HyperKZGHelpers.pre.sol
+            function univariate_group_evaluation(g_ptr, e, length, scratch) {
+                revert(0, 0)
+            }
+            // IMPORT-YUL HyperKZGHelpers.pre.sol
+            function compute_gl_msm(com_ptr, length, w_ptr, commitment_ptr, r, q, d, b, scratch) {
+                revert(0, 0)
+            }
+
+            function verify_hyperkzg(proof_ptr, transcript_ptr, commitment_ptr, x, y) {
+                function v_ptr(ptr, l) -> result {
+                    result := add(ptr, sub(mul(WORDX2_SIZE, l), WORDX2_SIZE))
+                }
+                function w_ptr(ptr, l) -> result {
+                    result := add(ptr, sub(mul(WORDX5_SIZE, l), WORDX2_SIZE))
+                }
+
+                let ell := mload(x)
+
+                // if ell == 0, then error
+                if iszero(ell) { err(ERR_HYPER_KZG_EMPTY_POINT) }
+
+                // Step 1: Run the transcript
```

Comment:
Maybe making `v_ptr(proof_ptr, ell)` and `w_ptr(proof_ptr, ell)` variables can save gas?